sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ProxyUtil.java

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.amqp.implementation;
+
+import com.azure.core.amqp.ProxyAuthenticationType;
+import com.azure.core.amqp.ProxyOptions;
+import com.azure.core.util.Configuration;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.logging.LogLevel;
+
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+/***
+ * Utility class to help with {@link ProxyOptions}.
+ */
+public final class ProxyUtil {
+
+    private static final Pattern HOST_PORT_PATTERN = Pattern.compile("^[^:]+:\\d+");
+    private static final ClientLogger LOGGER = new ClientLogger(ProxyUtil.class);
+
+    private ProxyUtil() {
+        // private default constructor
+    }
+
+    /***
+     * Static method to construct {@link ProxyOptions} using the values from configuration .
+     *
+     * @param configuration - The client library configuration
+     * @return the constructed {@link ProxyOptions} object
+     */
+    public static ProxyOptions getDefaultProxyConfiguration(Configuration configuration) {
```

Comment:
Change the method name to getProxyOptions()?  It looks not only return the default configuration.

Reply:
Oh, sorry. After checking EH and SB code, the name `getDefaultProxyConfiguration` is fine. Realize that you have combined `getDefaultProxyConfiguration` and `getProxyOptions` in one method.